build/ci/send-to-helix.yml

```diff
@@ -0,0 +1,58 @@
+# Please remember to update the documentation if you make changes to these parameters!
+parameters:
+  HelixSource: 'pr/default'              # required -- sources must start with pr/, official/, prodcon/, or agent/
+  HelixType: 'tests/default/'            # required -- Helix telemetry which identifies what type of data this is; should include "test" for clarity and must end in '/'
+  HelixBuild: $(Build.BuildNumber)       # required -- the build number Helix will use to identify this -- automatically set to the AzDO build number
+  HelixTargetQueues: ''                  # required -- semicolon delimited list of Helix queues to test on; see https://helix.dot.net/ for a list of queues
+  HelixAccessToken: ''                   # required -- access token to make Helix API requests; should be provided by the appropriate variable group
+  Configuration: 'Debug'
+  Architecture: 'x64'
+  HelixConfiguration: ''                 # optional -- additional property attached to a job
+  IncludeDotNetCli: true                 # optional -- true will download a version of the .NET CLI onto the Helix machine as a correlation payload; requires DotNetCliPackageType and DotNetCliVersion
+  EnableXUnitReporter: true              # optional -- true enables XUnit result reporting to Mission Control
+  WaitForWorkItemCompletion: true        # optional -- true will make the task wait until work items have been completed and fail the build if work items fail. False is "fire and forget."
+  HelixBaseUri: 'https://helix.dot.net/' # optional -- sets the Helix API base URI (allows targeting int)
+  Creator: 'ML.NET'                      # optional -- if the build is external, use this to specify who is sending the job
+  DisplayNamePrefix: 'Run Helix Tests'   # optional -- rename the beginning of the displayName of the steps in AzDO
+  continueOnError: false                 # optional -- determines whether to continue the build if the step errors; defaults to false
+  condition: succeeded()                 # optional -- condition for step to execute; defaults to succeeded()
+
+steps:
+  - powershell: '$env:Path = "$env:BUILD_SOURCESDIRECTORY\.dotnet;$env:Path"; powershell "$env:BUILD_SOURCESDIRECTORY\eng\common\msbuild.ps1 $env:BUILD_SOURCESDIRECTORY\eng\helix.proj /t:Test /bl:$env:BUILD_SOURCESDIRECTORY\artifacts\log\${{ parameters.Configuration }}\SendToHelix.binlog /p:Configuration=${{ parameters.Configuration }} /p:TargetArchitecture=${{ parameters.Architecture }} -warnAsError 0"'
+    displayName: ${{ parameters.DisplayNamePrefix }} (Windows)
+    env:
```

Comment:
When calling MSBuild using environment variables for the paramerters it makes mistakes harder to diagnose, consider using MSBuild parameters like in: https://github.com/dotnet/runtime/blob/b482c04080bb86f151e8fe3d6f2fe61649c1b5fa/eng/pipelines/libraries/helix.yml#L20-L30

Reply:
That line was taken directly from the `send-to-helix.yml` from Arcade. I don't know why they chose to do it that way, but my guess is that it makes it easier to try and debug locally. You can't run the yml file locally, but you can set up the env variable the same way so you can use the script the same way. I think again if you feel strongly its better the other way I'm happy to switch it, but otherwise I think I would rather keep it this way to be consistent with what Arcade has.

Reply:
I think it is easier to debug locally to copy a command rather than having to set multiple environment variables manually. Also when looking at binlogs we sometimes forget that environment variables come into play in MSBuild, so they are "obscure" for some people and harder to debug as the first thing and maybe only thing we look on a binlog are properties.